socorro/unittest/cron/crontabber_tests_base.py

```diff
@@ -7,72 +7,50 @@
 
 import datetime
 import json
-import os
 import unittest
-from past.builtins import basestring
-from collections import Sequence, Mapping, defaultdict
+from collections import defaultdict, Sequence
 
-import mock
-
-import configman
-from configman.dotdict import DotDictWithAcquisition
+from configman import ConfigurationManager
+from mock import Mock
+import six
 
 from socorro.cron.crontabber_app import CronTabberApp, JobStateDatabase
 
 
-environment = DotDictWithAcquisition(os.environ)
-environment.always_ignore_mismatches = True
+def get_config_manager(jobs=None, overrides=None):
+    crontabber_config = CronTabberApp.get_required_config()
+    crontabber_config.add_option('logger', default=Mock())
+    crontabber_config.add_option('metrics', default=Mock())
+
+    local_overrides = {}
+    if jobs:
+        local_overrides['crontabber.jobs'] = jobs
+
+    if isinstance(overrides, Sequence):
+        overrides.append(local_overrides)
+    elif overrides is not None:
+        overrides = [overrides, local_overrides]
+    else:
+        overrides = [local_overrides]
+
+    # Be sure to include defaults
+    overrides.insert(0, CronTabberApp.config_defaults)
+
+    return ConfigurationManager(
+        [crontabber_config],
+        values_source_list=overrides,
+        app_name='test-crontabber',
+        app_description='',
+        argv_source=[]
+    )
```

Comment:
We had a more complicated version of this in the code in 3 different places. I merged them all together into this one and reduced it down a little.

We could probably improve this further, but ultimately, I want to "rewrite" crontabber in Django so this will all go away.